### ISSUE TITLE
PyNE cannot write material objects to arbitrary locations

### ISSUE THREAD

**Issue by makeclean** [pyne/pyne#848]
It seems weird that we've had materials for so long and this hasn't come up, maybe I'm doing something dumb and someone will correct me.

Create a new material

```
  mat = Material({'H-1':1,'Fe-56':1},density = 1.0)
  mat.write_hdf5("my_file.h5",nucpath="/this/is/a/long/path/materials",nucid='/this/is/a/long/path/nucid')
```

Creates a new file, but there are no hdf5 dataspaces.

```
  /                        Group
```

The following works

```
  mat = Material({'H-1':1,'Fe-56':1},density = 1.0)
  mat.write_hdf5("my_file.h5")
```

Works, giving 

```
  /                        Group
  /materials               Dataset {1/Inf}
  /materials_metadata      Dataset {1/Inf}
  /nucid                   Dataset {2}
```

The following doesnt

```
  mat = Material({'H-1':1,'Fe-56':1},density = 1.0)
  mat.write_hdf5("my_file.h5",nucpath="/zuper/materials",nucid='/zuper/nucid')

  /                        Group
```

But, this does work

```
  mat = Material({'H-1':1,'Fe-56':1},density = 1.0)
  mat.write_hdf5("my_file.h5",nucpath="/materials",nucid='/nucid')
```

Giving

```
  /                        Group
  /materials               Dataset {1/Inf}
  /materials_metadata      Dataset {1/Inf}
  /nucid                   Dataset {2}
```

Am I doing something stupid, or should these all work?

**Comment by makeclean** [comment]
Looking at the tests, we dont have one for that so we should almost certainly add one anyhow, even if this isnt a bug, just in case :)

**Comment by makeclean** [comment]
This  might be a g++ thing, with GCC5 they appear to have broken the ABI https://gcc.gnu.org/gcc-5/changes.html#libstdcxx, which is incompatible with libraries compiled with pre v5 libraries. Im currently installing g++4.9 to test this hypothesis.

**Comment by gonuke** [comment]
This should be resolved by #1104
